### ISSUE TITLE
Current filename not cleared in parser

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#14658]
```
$ cat testparse.jl
str = """
function foo()
  1+1
end
"""
dump(Base.parse(str))
$ ~/julia-target/julia testparse.jl
Expr
  head: Symbol function
  args: Array(Any,(2,))
    1: Expr
      head: Symbol call
      args: Array(Any,(1,))
        1: Symbol foo
      typ: Any
    2: Expr
      head: Symbol block
      args: Array(Any,(2,))
        1: LineNumberNode
          file: Symbol /Users/kfischer/testparse.jl
          line: Int64 2
        2: Expr
          head: Symbol call
          args: Array(Any,(3,))
          typ: Any
      typ: Any
  typ: Any
```
Note the presence of a line number node, even though the parser has no idea where the string is from.

**Comment by JeffBezanson** [comment]
We should retire the `jl_parse_next` API. Probably makes sense to just parse a whole file at once.

**Comment by Keno** [comment]
I should also note that this is most likely a regression as the JuliaParser expect the filename to be cleared.

**Comment by Keno** [comment]
Do you have a quick workaround? I'd like to do some experiments with JuliaParser.jl and it would be good if I can get the tests passing first.

**Comment by ihnorton** [comment]
```
diff --git a/src/jlfrontend.scm b/src/jlfrontend.scm
index 41ecbab..04d83c9 100644
--- a/src/jlfrontend.scm
+++ b/src/jlfrontend.scm
@@ -132,6 +132,7 @@

 ;; parse only, returning end position, no expansion.
 (define (jl-parse-one-string s pos0 greedy)
+  (set! current-filename 'none)
   (let ((inp (open-input-string s)))
     (io.seek inp pos0)
     (let ((expr
```

**Comment by JeffBezanson** [comment]
Even better, wrap the body of `jl-parse-one-string` with `(with-bindings ((current-filename 'none))  ...  )`.

**Comment by JeffBezanson** [comment]
Should be fixed by #14707
